### ISSUE TITLE
Support Clipboard History

### ISSUE THREAD

**Issue by maxwxyz** [FreeCAD/FreeCAD#13416]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Problem description

From the FreeCAD Day 2024 Complaint Session:
**If you have copied multiple things in the past we want to be able to see, search and reuse that history.**

Currently you can copy and paste. In the clipboard you only get one item, which is the last thing you've copied. Other software has the ability to present you e.g. the past five that you have copied in the past to select from and paste.
If you have copied multiple things in the past, you want to be able to go back to the history and paste something other than the last thing you copied. For example, when you're copying the set of dimensions, you want to be able to go back to each one
Also for objects, caching copies of the copy.

### Full version info

```shell
0.22
```

### Subproject(s) affected?

Core

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by xtemp09** [comment]
I don't think it would be used by users given the soft supporting such a thing. Can clipboard history be an addon and installed via _Addon Manager_?
